packages/odf/components/s3-browser/buckets-list-page/bucketsListPage.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { odfDocLinks } from '@odf/shared/constants';
+import { LoadingBox } from '@odf/shared/generic';
+import { DOC_VERSION as odfDocVersion, useRefresh } from '@odf/shared/hooks';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { getValidFilteredData, ViewDocumentation } from '@odf/shared/utils';
+import {
+  ListPageBody,
+  ListPageCreateLink,
+  ListPageFilter,
+  ListPageHeader,
+  useListPageFilter,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { Trans } from 'react-i18next';
+import { Button, ButtonVariant, Grid, GridItem } from '@patternfly/react-core';
+import { SyncAltIcon } from '@patternfly/react-icons';
+import { BUCKET_CREATE_PAGE_PATH } from '../../../constants';
+import { BucketCrFormat } from '../../../types';
+import { NoobaaS3Provider } from '../noobaa-context';
+import { BucketsListTable } from './bucketListTable';
+import { BucketPagination } from './bucketPagination';
+
+const BucketsListPageBody: React.FC = () => {
+  const { t } = useCustomTranslation();
+  const [fresh, triggerRefresh] = useRefresh();
+  const [bucketInfo, setBucketInfo] = React.useState<
+    [BucketCrFormat[], boolean, any]
+  >([[], false, undefined]);
+  const [buckets, loaded, loadError] = bucketInfo;
+  const [allBuckets, filteredBuckets, onFilterChange] =
+    useListPageFilter(buckets);
+
+  return (
+    <ListPageBody>
+      <Grid>
+        <GridItem md={11} sm={12} className="pf-v5-u-mt-md">
+          <div className="pf-v5-u-display-flex pf-v5-u-flex-direction-column pf-v5-u-flex-direction-row-on-md">
+            <ListPageFilter
+              loaded={true}
+              hideLabelFilter={true}
+              nameFilterPlaceholder={t('Search a bucket by name')}
+              data={getValidFilteredData(allBuckets)}
+              onFilterChange={onFilterChange}
+            />
+            <div className="pf-v5-u-ml-md pf-v5-u-mt-md">
+              <Button
+                data-test="bucket-list-sync-button"
+                variant={ButtonVariant.plain}
+                onClick={triggerRefresh}
+                isDisabled={!fresh}
+              >
+                <SyncAltIcon />
+              </Button>
+            </div>
+          </div>
+        </GridItem>
+        <GridItem md={1} sm={12}>
+          <div className="pf-v5-u-mt-md">
+            {fresh && <BucketPagination setBucketInfo={setBucketInfo} />}
+          </div>
+        </GridItem>
+      </Grid>
+      {fresh ? (
+        <BucketsListTable
+          allBuckets={allBuckets}
+          filteredBuckets={filteredBuckets}
+          loaded={loaded}
+          error={loadError}
+        />
+      ) : (
+        <LoadingBox />
+      )}
```

Comment:
nits (optional comment): `StatusBox` inside `BucketsListTable` should handle the loading state in form of table skeleton...
```suggestion
      {fresh && (
        <BucketsListTable
          allBuckets={allBuckets}
          filteredBuckets={filteredBuckets}
          loaded={loaded}
          error={loadError}
        />
      )}
```